src/Sfa.Das.ApprenticeshipInfoService.Infrastructure/Elasticsearch/ApprenticeshipSearchServiceV2.cs

```diff
@@ -0,0 +1,276 @@
+                .Bool(mb1 => mb1
+                    .Should(
+                        bs1 => bs1
+                                   .DateRange(r => r
+                                       .GreaterThanOrEquals(DateTime.Today)
+                                       .Field(f => f.LastDateForNewStarts))
+                               || bs1
+                                   .Bool(bsb1 => bsb1
+                                       .MustNot(mn => mn
+                                           .Exists(e => e
+                                               .Field(f => f.LastDateForNewStarts))))));
+        }
+
+        private static Func<QueryContainerDescriptor<ApprenticeshipSearchResultsItem>, QueryContainer> PublishedApprenticeship()
+        {
+            return f => f
+                .Term(t => t
+                    .Field(fi => fi.Published).Value(true));
+        }
+
+        private static Func<QueryContainerDescriptor<ApprenticeshipSearchResultsItem>, QueryContainer> MustBeNonExpiredApprenticceship()
+        {
+            return m1 => m1
+                .Bool(mb1 => mb1
+                    .Should(
+                        bs1 => bs1
+                                   .DateRange(r => r
+                                       .GreaterThanOrEquals(DateTime.Now)
+                                       .Field(f => f.EffectiveTo))
+                               || bs1
+                                   .Bool(bsb1 => bsb1
+                                       .MustNot(mn => mn
+                                           .Exists(e => e
+                                               .Field(f => f.EffectiveTo))))));
+        }
+
+        private static Func<QueryContainerDescriptor<ApprenticeshipSearchResultsItem>, QueryContainer> MustBeStartedApprenticeship()
+        {
+            return m0 => m0
+                .Bool(mb0 => mb0
+                    .Should(
+                        s0 => s0
+                                  .DateRange(r => r
+                                      .LessThanOrEquals(DateTime.Now)
+                                      .Field(f => f.EffectiveFrom))
+                              || s0
+                                  .Bool(b2 => b2
+                                      .MustNot(mn => mn
+                                          .Exists(e => e
+                                              .Field(f => f.EffectiveFrom))))));
+        }
+
+        private SearchDescriptor<ApprenticeshipSearchResultsItem> GetSearchDescriptor(
+            int page, int take, string formattedKeywords, int order, IEnumerable<int> selectedLevels)
+        {
+            return formattedKeywords == "*"
+                ? GetAllSearchDescriptor(page, take, formattedKeywords, order, selectedLevels)
+                : GetKeywordSearchDescriptor(page, take, formattedKeywords, order, selectedLevels);
+        }
+
+        private SearchDescriptor<ApprenticeshipSearchResultsItem> GetAllSearchDescriptor(
```

Comment:
It was mentioned about moving these queries to elastic search, would this be a good time to do this also rather than in code?

Reply:
Yeah. Not for this story. A much bigger task.